subsys/mgmt/osdp/src/osdp_sc.c

```diff
@@ -222,16 +221,17 @@ int osdp_compute_mac(struct osdp_pd *pd, int is_cmd,
 	return 0;
 }
 
-void osdp_sc_init(struct osdp_pd *pd)
+void osdp_sc_setup(struct osdp_pd *pd)
 {
-	uint8_t key[16];
+	uint8_t scbk[16];
+	bool preserve_scbk = is_pd_mode(pd) || ISSET_FLAG(pd, PD_FLAG_HAS_SCBK);
 
-	if (ISSET_FLAG(pd, PD_FLAG_PD_MODE)) {
-		memcpy(key, pd->sc.scbk, 16);
+	if (preserve_scbk) {
+		memcpy(scbk, pd->sc.scbk, 16);
```

Comment:
I agree with your point.. but it feels like that's not the kind of threat model we work with here(?). Anyhow, I cannot find a secure variant of of memset in zephyr with my grep-ing so I wonder what the others are doing about this..

Reply:
@carlescufi @gmarull @erwango thoughts?

Reply:
> but it feels like that's not the kind of threat model we work with here(?). Anyhow, I cannot find a secure variant of of memset in zephyr with my grep-ing so I wonder what the others are doing about this..

Usually this is handled by crypto library (i.e. mbedTLS clears allocated memory before feeeing it), so in Zephyr code this kind of functionality is not usual.
In this case threat comes not from hostile memory access i.e. from another thread but from accidential disclosure of memory content (i.e. by dump logging in non related thread), so witch such assumption regular memset is good enough.

Reply:
> so in Zephyr code this kind of functionality is not usual.

Possible. But I'd expect there to be other modules/subsystem where we invoke low level crypto APIs. To be fair, I haven't spent time looking for it seriously I just grep-ed for varying call sites for memset such as `*_memset_*`. I'll allocate some time for this; if I cannot find one, I'll add it.

> So witch such assumption regular memset is good enough.

No, regular memset gets optimized away by compilers with aggressive optimization levels/options.

Reply:
So we have options:
- ignore problem (leaving master password in memory is for our product a no go)
- add something like osdp_memset_s function (i.e. macro which will use standard memset_s in case of C11 compilation or something like in mbedtls_platform_zeroize otherwise)
- use crypto API which will allow to avoid using plain passwords for secure session setup (a little different topic than using plain password for KEYSET command)